File: Java Programs/1.4.2 GuessingGame.java
/*
 # - - - - - - - - - - - - - - - - #
 # SummerTech/Coditum Curriculum   #
 # Created by Sam Perlman          #
 #                                 #
 # Report any bugs, issues, or     #
 # corrections to:                 #
 # matt@summertech.net             #
 #                                 #
 #                                 #
 # Module 1.4.2                    #
 # - - - - - - - - - - - - - - - - #
 
 SUMMERTECH CONFIDENTIAL
 COPYRIGHT (C) 2017 SUMMERTECH, INC. ALL RIGHTS RESERVED
 */

import java.util.Random;
import java.util.Scanner;

public class GuessingGame {

	public static void main(String[] args) {

    Scanner scan = new Scanner(System.in);
		Random rand = new Random();
		
		int target = rand.nextInt(100) + 1;
		int guess = -1;
		
		int numGuesses = 0;
		
		while (guess != target) {
			System.out.println("Guess a number between 1 and 100 inclusive!");
			guess = scan.nextInt();
			numGuesses++;
			
			if (guess > target) {
				System.out.println("Whoops. Too high!");
			}
			else if (guess < target) {
				System.out.println("Too bad. You guessed too low!");
			}
		}
    
		System.out.println("You guessed it! It only took you " + numGuesses + " tries!");
	}
}

/* Challenge: 
    - Have the student make the game stop after a certain number of guesses
      by adding to the condition of the while loop.
*/

File: Python Programs/1.4.2 GuessingGame.py
# - - - - - - - - - - - - - - - - #
# SummerTech/Coditum Curriculum   #
# Created by Matthew Baptist      #
#                                 #
# Report any bugs, issues, or     #
# corrections to:                 #
# matt@summertech.net             #
#                                 #
#                                 #
# Module 1.4.2                    #
# - - - - - - - - - - - - - - - - #

#SUMMERTECH CONFIDENTIAL
#COPYRIGHT (C) 2017 SUMMERTECH, INC. ALL RIGHTS RESERVED

"""
Teaching Notes

Goal: Have the student think through how to solve a problem with while loops. Introduce the syntax for finding a random number.

Remarks:

1. 


"""

from random import randint

x = randint(1, 100)

guess = -1
tries = 0

while not guess == x:

  guess = int(input("Please enter a number between 1-100. "))
  tries += 1

  if guess > x:
    print("Too high! ", end = "")

  if guess < x:
    print("Too low! ", end = "")

  if tries == 7 and not guess == x:
    print("You're out of tries, you lose.")
    break
  elif not guess == x:
    print("Try again.")

if guess == x:
  print("You won!")
